gno.land/pkg/sdk/vm/msgs.go

```diff
@@ -79,6 +88,59 @@ func (msg MsgAddPackage) GetReceived() std.Coins {
 	return msg.Deposit
 }
 
+//----------------------------------------
+// MsgEval
+
+// MsgEval - eval a Gno Expr.
+type MsgEval struct {
+	PkgPath string         `json:"pkg_path" yaml:"pkg_path"`
+	Expr    string         `json:"expr" yaml:"expr"`
+	Caller  crypto.Address `json:"caller" yaml:"caller"`
+
+	// XXX: This field is experimental, use with care as output is likely to change
+	// Default format is machine
+	Format Format `json:"format" yaml:"format"`
+}
+
+var _ std.Msg = MsgEval{}
+
+func NewMsgEval(format Format, pkgPath, expr string) MsgEval {
```

Comment:
```suggestion
func NewMsgEval(pkgPath, expr string, format Format) MsgEval {
```

Personal preference for this order.

Reply:
the more i think about it, the more i'm for having such struct, but unexported, without New constructor, not in the msgs.go file.

and probably centralizing everything about the query in query.go.